docs/proposals/karmada-operator/pod-disruption-budget/README.md

```diff
@@ -0,0 +1,160 @@
+    // +optional
+    MaxUnavailable *intstr.IntOrString `json:"maxUnavailable,omitempty"`
+}
+```
+---
+
+## User Stories
+
+1. **High Availability During Planned Disruptions**  
+   As a cloud infrastructure engineer managing multiple clusters with Karmada control planes, I need to ensure that voluntary disruption, such as node drains or rolling upgrades, do not evict too many replicas at once. By defining PodDisruptionBudgets through the operator, I can maintain control plane availability and keep workloads across member clusters in sync, even during maintenance windows.
+
+2. **Consistent Experience Across Installation Methods**  
+   As a cloud infrastructure engineer responsible for deploying Karmada in various ways, I want the same fine-grained control over PodDisruptionBudgets whether I'm using Helm or the Karmada Operator. This consistency simplifies my workflow, prevents confusion, and lets me standardize on a single set of eviction policies.
+
+3. **Improved Resilience for Production Environments**  
+   As a cloud infrastructure engineer working under strict Service Level Agreements (SLAs), I cannot afford unplanned downtime in the Karmada control plane. PDBs are vital for graceful rolling updates and node maintenance, and having them supported natively by the operator ensures the control plane remains stable and resilient in production scenarios.
+
+4. **Alignment with Organizational Policies**  
+   As a cloud infrastructure engineer in an organization with strict compliance standards, I need to enforce PodDisruptionBudgets for critical workloads. Integrating PDB support directly into the operator makes it straightforward to adhere to these policies, eliminating the risk of manual misconfiguration and ensuring compliance from the start.
+
+5. **Avoiding Manual Workarounds**  
+   As a cloud infrastructure engineer provisioning Karmada with the operator, I currently have to create PDBs manually afterward, which is easy to overlook or configure incorrectly. By having first-class PDB support, I can rely on the operator to automatically set up disruption budgets, guaranteeing my control plane meets high-availability requirements from day one.
+
+---
+
+## Risks and Mitigations
+
+### Misconfiguration of PDB
+- If a user sets `minAvailable` higher than the number of replicas, the pods could become unschedulable during updates or evictions.  
+  **Mitigation**: The operator could report and error during validation of the Karmada resource if `replicas < minAvailable`.
+
+### Backward Compatibility
+- Existing installations don’t create or expect PDBs.  
+  **Mitigation**: The new fields are optional; if not set, no PDB is created, preserving current behavior.
+
+---
+
+## Design Details
+
+### Workflow
+
+#### Reconcile Logic
+- For each control plane component, the operator checks if `podDisruptionBudgetConfig` is set.
+    - If set, the operator constructs or updates a `PodDisruptionBudget` object using the provided values.
```

Comment:
I wonder how the `PodDisruptionBudget` object can be removed while removing the Karmada instance.
Set an owner reference to the object?

Reply:
> I wonder how the `PodDisruptionBudget` object can be removed while removing the Karmada instance. Set an owner reference to the object?

@RainbowMango , great question!

And yes. Precisely. We'll create it with an owner reference to ensure it's properly garbaged collected on deletion.

Reply:
make sense.